### ISSUE TITLE
`BackendConnector` types

### ISSUE THREAD

**Issue by 2Pacalypse-** [i18next/i18next#1872]
## 🚀 Feature Proposal

Currently, the `BackendConnector` is typed as `any`: https://github.com/i18next/i18next/blob/master/index.d.ts#L982
It would be nice to have types for this.

## Motivation

When using the "save missing keys" feature and handling the backend request yourself, you need to use the backend connector's `saveMissing` method. It would be nice to have some type safety there.

## Example

```js
const connector = i18next.services.backendConnector
if (connector) {
  for (const key in ctx.request.body) {
    connector.saveMissing([lng], ns, key, ctx.request.body[key])
  }
}
```

**Comment by jamuhl** [comment]
just my opinion...the backendConnector is internal magic to glue backend implementations to core functionality...i don't see a use case for using it inside your code:

-> saveMissing basically just calls the backend implementation: https://github.com/i18next/i18next/blob/master/src/BackendConnector.js#L234

**Comment by 2Pacalypse-** [comment]
Yeah, I saw that I basically don't even need the `i18next` for handling the "saveMissing" requests on the backend, I just need the functionality of the backend (`i18next-fs-backend` in my case).

However, I wasn't quite sure how to use the FsBackend on its own. For example, I wasn't sure what the `services` param in its constructor is or how to provide it to the backend: https://github.com/i18next/i18next-fs-backend/blob/master/lib/index.js#L17

That's why I went with the BackendConnector route, but if you have an idea on how I can use the `i18next-fs-backend` directly, without going through the core `i18next`, please let me know.

**Comment by jamuhl** [comment]
not sure why you call saveMissing yourself -> i18next does that for you when it gets called with unknown keys during runtime

**Comment by adrai** [comment]
You best use [i18next-http-middleware](https://github.com/i18next/i18next-http-middleware)... example: https://github.com/i18next/i18next-http-middleware/blob/master/example/basic/index.js

**Comment by jamuhl** [comment]
sorry...missed the point you're using it in the backend -> https://github.com/i18next/i18next-http-middleware/blob/master/lib/index.js#L234 does the same...but using that middleware instead of building something on your own might be more future proof

**Comment by 2Pacalypse-** [comment]
Yes, I've looked at the `i18next-http-middleware` before (that's where I got the usage of `backendConnector` from :p). However, I've had issues making it work with Koa, and in the end decided it's an overkill for something that amounts to few lines of code to handle myself.

So to go back to the original issue, I still think it's a good idea to add types to the `BackendConnector` class, or maybe extracting the "saveMissing" feature somehow so it can be used without `backendConnector`.
